metaserver/src/main/scala/scala/meta/languageserver/providers/DocumentFormattingProvider.scala

```diff
@@ -2,25 +2,40 @@ package scala.meta.languageserver.providers
 
 import java.nio.file.Files
 import scala.meta.languageserver.Formatter
+import scala.meta.languageserver.Configuration
 import com.typesafe.scalalogging.LazyLogging
 import langserver.messages.DocumentFormattingResult
 import langserver.types.Position
 import langserver.types.Range
 import langserver.types.TextEdit
 import org.langmeta.inputs.Input
 import org.langmeta.io.AbsolutePath
+import monix.reactive.Observable
+import monix.eval.Task
+
+class DocumentFormattingProvider(configuration: Observable[Configuration], cwd: AbsolutePath) extends LazyLogging {
+
+  private lazy val scalafmt = Formatter.classloadScalafmt("1.3.0")
+
+  private def formatterFromConfiguration: Task[Formatter] = configuration.lastL.map { conf =>
+    if (conf.scalafmt.enable) scalafmt
```

Comment:
I opened https://github.com/scalameta/language-server/issues/159 to track making the scalafmt version configurable. With `configuration.distinctByKey(_.scalafmt)` we can avoid re-classloading on every config change.